src/addons/addonmessage.cpp

```diff
@@ -76,35 +78,45 @@ AddonMessage::AddonMessage(QObject* parent, const QString& manifestFileName,
 
 AddonMessage::~AddonMessage() { MVPN_COUNT_DTOR(AddonMessage); }
 
-void AddonMessage::dismiss() {
-  m_dismissed = true;
-  disable();
-
+// static
+AddonMessage::State AddonMessage::loadMessageState(const QString& id) {
   SettingsHolder* settingsHolder = SettingsHolder::instance();
-  Q_ASSERT(settingsHolder);
 
-  QStringList dismissedAddonMessages = settingsHolder->dismissedAddonMessages();
-  dismissedAddonMessages.append(id());
-  settingsHolder->setDismissedAddonMessages(dismissedAddonMessages);
-}
+  bool isValidState;
+  QString stateSetting = settingsHolder->getAddonSetting(StateQuery(id));
+  QMetaEnum stateMetaEnum = QMetaEnum::fromType<State>();
 
-void AddonMessage::maskAsRead() {
-  if (m_isRead) {
-    return;
+  if (!stateSetting.isEmpty()) {
```

Comment:
what happens if we remove check? Would the code work?

Reply:
Hm, I guess we can remove it since 1. there is a default value, so this should neve really be empty 2. isValidState will be `false` if it's empty.

Reply:
I will just add two tests cases for that in case I am wrong.